### DISCUSSION TITLE
Regarding the issue of logging in and logging out.

### DISCUSSION THREAD

**Discussion by wyhdil** [eclipse-theia/theia-cloud#262] [Q&A]
Hello, I've encountered some strange issues related to the user logout functionality. I hope you can help.

I've used the 'try-now-page' as the login module. This is supposed to trigger the user logout function, so I plan to test whether the logout feature is effective.
![e34e735dffa322faea2156aa0c391624](https://github.com/eclipsesource/theia-cloud/assets/69418800/a82db5b0-c5b2-4aa7-924c-255faa871258)

From the code, it can be seen that the URL redirection for the logout functionality is obtained from `keycloak.createLogoutUrl().`

![img_v3_0261_7ef6b40d-cf60-4c2b-8846-547ffeda81eg](https://github.com/eclipsesource/theia-cloud/assets/69418800/c57798ac-897b-4488-b413-10792485622b)

So, I obtained the URL for logout through the console.
![82f28728b05b7ac7a2a7cc5935d02299](https://github.com/eclipsesource/theia-cloud/assets/69418800/412e30bf-def6-49ff-9cbb-edd572163e40)

And then I started testing the logout functionality: I logged in normally, opened the IDE. Then, in another browser window, I entered the logout URL, which redirected to a Keycloak page. 
![img_v3_0261_1d6fa6d3-d6cb-4fbc-9447-36174a5aeccg](https://github.com/eclipsesource/theia-cloud/assets/69418800/032798a5-2a6f-4dec-85cb-bc9a458aaa83)

I clicked on logout. However, when I refreshed and tried to access the previously copied IDE URL again, the route was still accessible. 
This is obviously incorrect. In theory, after logging out, I shouldn't be able to open the IDE using the previously copied URL.

So, I would like to know if there is an issue with the logout functionality.Thanks!

**Comment by jfaltermeier** [comment]
Hi, the log out on the try now page only logs the user out of the Github authentication provider. 
The session cookie for the IDE URL, which is handled by the oauth2-proxy, is not deleted. For our try now demo use case this is fine. 

However I agree that we should create an example/documentation that shows how to implement the logout. I've created https://github.com/eclipsesource/theia-cloud/issues/264 for this. 
The documentation for the proxy we use can be found here: https://oauth2-proxy.github.io/oauth2-proxy/docs/features/endpoints/#sign-out
